common/settings/settings.ts

```diff
@@ -30,6 +30,86 @@ export interface MiscSettings {
     readonly pauseOnHoverMode: PauseOnHoverMode;
 }
 
+/*
+These are all the possible scenarios which can result in a match. We don't need to support every possible combination,
+as some are not useful or inconsistent. Inconsistent meaning the order the user collects forms affects what future forms
+are considered collected (e.g collecting the lemma will match all forms but user needs to collect every inflection if
+they don't ever collect the lemma).
+
+Lemma In Subtitle
+-----------------------------------------------------------------
+| User Collection | LEMMA_FORM_COLLECTED | EXACT_FORM_COLLECTED |
+-----------------------------------------------------------------
+| Lemma           |         MATCH        |         MATCH        |
+| Inflection      |          NO          |          NO          |
+-----------------------------------------------------------------
+
+Inflection In Subtitle
+-----------------------------------------------------------------
+| User Collection | LEMMA_FORM_COLLECTED | EXACT_FORM_COLLECTED |
+-----------------------------------------------------------------
+| Lemma           |         MATCH        |          NO          |
+| Same Inflection |          NO          |         MATCH        |
+| Diff Inflection |          NO          |          NO          |
+-----------------------------------------------------------------
+*/
+export enum TokenMatchStrategy {
+    ANY_FORM_COLLECTED = 'ANY_FORM_COLLECTED', // All scenarios above result in MATCH
+    LEMMA_OR_EXACT_FORM_COLLECTED = 'LEMMA_OR_EXACT_FORM_COLLECTED', // See LEMMA_FORM_COLLECTED and EXACT_FORM_COLLECTED columns above
+    LEMMA_FORM_COLLECTED = 'LEMMA_FORM_COLLECTED', // See LEMMA_FORM_COLLECTED column above
+    EXACT_FORM_COLLECTED = 'EXACT_FORM_COLLECTED', // See EXACT_FORM_COLLECTED column above
+}
+
+export enum TokenMatchStrategyPriority {
+    EXACT = 'EXACT',
+    LEMMA = 'LEMMA',
+    BEST_KNOWN = 'BEST_KNOWN',
+    LEAST_KNOWN = 'LEAST_KNOWN',
+}
+
+export enum TokenStyling {
+    TEXT = 'TEXT',
+    BACKGROUND = 'BACKGROUND',
+    UNDERLINE = 'UNDERLINE',
+    OVERLINE = 'OVERLINE',
+    OUTLINE = 'OUTLINE',
+}
+
+export enum TokenStatus {
+    UNCOLLECTED = 0,
+    UNKNOWN = 1,
+    LEARNING = 2,
+    GRADUATED = 3,
+    YOUNG = 4,
+    MATURE = 5, // If ever adding more statuses, they should go after MATURE and getFullyKnownTokenStatus should be updated
+}
+
+export function getFullyKnownTokenStatus(): TokenStatus {
+    return TokenStatus.MATURE; // If future statuses are optional, this logic may need to change
+}
+
+export interface DictionaryTrack {
+    readonly dictionaryColorizeSubtitles: boolean;
+    readonly dictionaryTokenMatchStrategy: TokenMatchStrategy;
+    readonly dictionaryTokenMatchStrategyPriority: TokenMatchStrategyPriority;
+    readonly dictionaryYomitanUrl: string;
+    readonly dictionaryYomitanScanLength: number;
+    readonly dictionaryAnkiEnabled: boolean;
```

Comment:
I think we should just have a single boolean for now `dictionaryColorizeSubtitles` that enables or disables coloring. I don't think we gain anything right now by having a separate boolean for `dictionaryAnkiEnabled`, as there's no new user-facing behavior unless both booleans are enabled. I understand the desire for flexibility and the potential to support different backends for learning status in the future, but even if we did that it would be trivial to add a new setting at that point in time.